### ISSUE TITLE
DocumentArray `sort` should also get `top_k` to allow for more efficient sorting

### ISSUE THREAD

**Issue by JoanFM** [jina-ai/jina#3467]
**Describe the feature**
Right now `sort` is done on all the `Document` collection, I would like to sort only up to a `top_k`.

**Comment by sauravgarg540** [comment]
Hey @JoanFM, I would like to work on this issue.

**Comment by JoanFM** [comment]
Sure @sauravgarg540 ! Just ask if u need any help!

**Comment by sauravgarg540** [comment]
Hey @JoanFM, I have drafted a [PR](https://github.com/jina-ai/jina/pull/3468). Waiting for your feedback.

**Comment by sauravgarg540** [comment]
Hey @JoanFM, I have a doubt regarding the aim of this issue, it would be great if you can clarfy it for me

- while sorting in ascending order, the implementation should get the `top_k`  smallest values sort them and keep it at the top while the rest of the list may not be sorted.

Did I get the issue right?

**Comment by JoanFM** [comment]
Yes, this is the idea, and for descending it applies the other way around
